diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff89633..3526045 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,7 +10,11 @@ add_definitions(-DFLASHMQ_VERSION=\"${PROJECT_VERSION}\")
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-SET(CMAKE_CXX_FLAGS "-msse4.2")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}")
+find_package(SSE)
+if (SSE_FOUND)
+    SET(CMAKE_CXX_FLAGS "-msse4.2")
+endif()
 
 check_cxx_compiler_flag("-rdynamic" COMPILER_SUPPORTS_RDYNAMIC)
 if (${COMPILER_SUPPORTS_RDYNAMIC})
diff --git a/FindSSE.cmake b/FindSSE.cmake
new file mode 100644
index 0000000..1fd7e75
--- /dev/null
+++ b/FindSSE.cmake
@@ -0,0 +1,15 @@
+include(CheckCXXSourceCompiles)
+set(CMAKE_REQUIRED_FLAGS "")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    set(CMAKE_REQUIRED_FLAGS "-msse4.2")
+endif()
+
+check_cxx_source_compiles("
+    #include <xmmintrin.h>
+    #include <smmintrin.h>
+    #include <tmmintrin.h>
+    int main() { return 0; }"
+    SSE_FOUND
+)
+
